### PR TITLE
[RFF] "NIST OSCAL Umbrella of Projects" Page

### LINKED ISSUE #1649: Is there documentation describing the different NIST OSCAL repos?
### Question

Is there any description of what the different OSCAL repositories are and which one should be used for what?

### PR DESCRIPTION
**Pull request by nikitawootten-nist**
# Committer Notes

Fixes #1649

Change preview:

image

### All Submissions:

- [ ] Have you selected the correct base branch per [Contributing](https://github.com/usnistgov/OSCAL/blob/main/CONTRIBUTING.md) guidance?
- [ ] Have you set "[Allow edits and access to secrets by maintainers
](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Do all automated CI/CD checks pass?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes you made? Changes to the OSCAL website can be made in the docs/content directory of your branch.

### CODE REVIEW

File: docs/content/about/projects/_index.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+### `liboscal-java`
+
+`liboscal-java` is a suite of Java libraries based on [Metaschema-Java](#Metaschema-Java) that implement common OSCAL operations such as reading/writing artifacts, profile resolution, and validation.
+
+Repository: https://github.com/usnistgov/liboscal-java
+
+### OSCAL-CLI
+
+OSCAL-CLI provides an easy-to-use CLI based on [`liboscal-java`](#liboscal-java).
```
Comment (Compton-NIST) [inline docs/content/about/projects/_index.md:69]:
```suggestion
OSCAL-CLI provides an easy-to-use command-line interface (CLI) based on [`liboscal-java`](#liboscal-java). 
```

File: docs/content/about/projects/_index.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+Metaschema is the modeling language that OSCAL uses to define all models. This repository documents Metaschema and provides XML pipelines used by OSCAL to perform various operations, like generation of model documentation, schemas, and content converters.
+
+Repository: https://github.com/usnistgov/metaschema
+
+### Metaschema-Java
+
+Metaschema-Java is the Java implementation of Metaschema, providing the toolchain used by [`liboscal-java`](#liboscal-java) to parse and validate OSCAL artifacts.
+
+Repository: https://github.com/usnistgov/metaschema-java
```
Comment (iMichaela) [inline docs/content/about/projects/_index.md:93]: suggestion: Repository: [https://github.com/usnistgov/metaschema-java](https://github.com/usnistgov/metaschema-java)

File: docs/content/about/projects/_index.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+### Hugo-USWDS
+
+Both the OSCAL Project and the [Metaschema](https://pages.nist.gov/metaschema/) share this repository as a common website template based on the [US Web Design System](https://designsystem.digital.gov/).
+
+Repository: https://github.com/usnistgov/hugo-uswds
```
Comment (iMichaela) [inline docs/content/about/projects/_index.md:99]: suggestion: Repository: [https://github.com/usnistgov/hugo-uswds](https://github.com/usnistgov/hugo-uswds)

File: docs/content/about/projects/_index.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+## NIST-Maintained Tools
```
Comment (iMichaela) [inline docs/content/about/projects/_index.md:101]: suggestion: ## Other OSCAL-Related Tools

File: docs/content/about/projects/_index.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+These projects are run by the OSCAL team, but may not be the primary focus of the NIST OSCAL team.
```
Comment (iMichaela) [inline docs/content/about/projects/_index.md:103]: These projects are managed by the NIST OSCAL team, but, at times, might be given a lower priority than the core OSCAL projects.

File: docs/content/about/projects/_index.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+### OSCAL-Deep-Diff
+
+A schema-agnostic diffing tool and library built to compare OSCAL JSON documents.
+
+Repository: https://github.com/usnistgov/oscal-deep-diff
```
Comment (iMichaela) [inline docs/content/about/projects/_index.md:109]: suggestion: Repository: [https://github.com/usnistgov/oscal-deep-diff](https://github.com/usnistgov/oscal-deep-diff)

File: docs/content/about/projects/_index.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+### OSCAL-XSLT
+
+This repository contains a collection of XSLT-based utilities for displaying and processing OSCAL documents.
+
+Repository: https://github.com/usnistgov/oscal-xslt
```
Comment (iMichaela) [inline docs/content/about/projects/_index.md:115]: suggestion: Repository: [https://github.com/usnistgov/oscal-xslt](https://github.com/usnistgov/oscal-xslt)

File: docs/content/about/projects/_index.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+## Other Projects
+
+### What about projects not on this list?
+
+NIST OSCAL Team staff, or NIST staff more generally, develop software for their own education or experimentation. If you see NIST staff, either on the [`usnistgov` GitHub organization](https://github.com/usnistgov/) or user accounts associated with official NIST work, these are not considered official NIST projects until they are included in the categories above. Individuals may pursue work on them, but they have no official maintenance or support guarantees from this team.
```
Comment (iMichaela) [inline docs/content/about/projects/_index.md:121]: **Typo**: When two or more nouns are connected using or, we use a singular verb with singular nouns and a plural verb with plural countable nouns. Staff in American English is considered singular (one group of people). In British English 'staff' is considered a plural noun. 
Please replace "develop software" with "develop**s** software."

**Review by iMichaela — CHANGES_REQUESTED** [review]
Please see the suggestions provided and the typo that needs correction. Thank you.

File: docs/content/about/projects/_index.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+heading: OSCAL Project Structure
+weight: 20
+toc:
+  enabled: true
+---
+
+The OSCAL project is made up of several components, each of which lives in its own repository on GitHub under the [`usnistgov`](https://github.com/usnistgov) organization.
+
+These repositories form a complex patchwork of projects that depend on one-another:
+
```
Comment (iMichaela) [inline docs/content/about/projects/_index.md:12]: The Mermaid graph will need updates if you accept any of the suggestions related to the subtitles of the page.

File: docs/content/about/projects/_index.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+### OSCAL
+
+The OSCAL repository is the largest and most complex repository in the OSCAL project. It contains:
+- The [Metaschema](#Metaschema) model definitions that define OSCAL
+- The OSCAL website along with all tutorials and model documentation
+- Informal specifications for operations on OSCAL models (such as [profile resolution](https://pages.nist.gov/OSCAL/concepts/processing/profile-resolution/))
+- XML and JSON schemas for all models
+- XML transformation pipelines for xml/json conversion
+
+Repository: https://github.com/usnistgov/OSCAL
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:59]: Markdown renders URLs as hyperlinks by default.

File: docs/content/about/projects/_index.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+
+`liboscal-java` is a suite of Java libraries based on [Metaschema-Java](#Metaschema-Java) that implement common OSCAL operations such as reading/writing artifacts, profile resolution, and validation.
+
+Repository: https://github.com/usnistgov/liboscal-java
+
+### OSCAL-CLI
+
+OSCAL-CLI provides an easy-to-use CLI based on [`liboscal-java`](#liboscal-java). 
+
+Repository: https://github.com/usnistgov/oscal-cli
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:71]: Markdown renders URLs as hyperlinks by default.

File: docs/content/about/projects/_index.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+
+### OSCAL-content
+
+The OSCAL content repository provides a library of official NIST-maintained OSCAL examples in XML, JSON, and YAML.
+
+Repository: https://github.com/usnistgov/oscal-content
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:77]: Markdown renders URLs as hyperlinks by default.

File: docs/content/about/projects/_index.md [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+
+## NIST Dependencies
+
+These NIST projects underpin the OSCAL project, but are governed separately from the OSCAL umbrella.
+
+### Metaschema
+
+Metaschema is the modeling language that OSCAL uses to define all models. This repository documents Metaschema and provides XML pipelines used by OSCAL to perform various operations, like generation of model documentation, schemas, and content converters.
+
+Repository: https://github.com/usnistgov/metaschema
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:87]: Markdown renders URLs as hyperlinks by default.

File: docs/content/about/projects/_index.md (hunk 6 above)
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:109]: Markdown renders URLs as hyperlinks by default.

File: docs/content/about/projects/_index.md (hunk 7 above)
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:115]: Markdown renders URLs as hyperlinks by default.

File: docs/content/about/projects/_index.md [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+- XML and JSON schemas for all models
+- XML transformation pipelines for xml/json conversion
+
+Repository: https://github.com/usnistgov/OSCAL
+
+### `liboscal-java`
+
+`liboscal-java` is a suite of Java libraries based on [Metaschema-Java](#Metaschema-Java) that implement common OSCAL operations such as reading/writing artifacts, profile resolution, and validation.
+
+Repository: https://github.com/usnistgov/liboscal-java
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:65]: Markdown renders URLs as hyperlinks by default.

**Comment by wendellpiez** [comment]
Looks good. Wondering if a top-down view would be clearer than left-right.

File: docs/content/about/projects/_index.md (hunk 9 above)
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:12]: Resolved in https://github.com/usnistgov/OSCAL/pull/1702/commits/1a4914c840bfc5fd42ad873a2cd0c66bbf570c7a

File: docs/content/about/projects/_index.md (hunk 8 above)
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md]: Resolved in https://github.com/usnistgov/OSCAL/pull/1702/commits/1a4914c840bfc5fd42ad873a2cd0c66bbf570c7a

File: docs/content/about/projects/_index.md (hunk 5 above)
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md]: Resolved in https://github.com/usnistgov/OSCAL/pull/1702/commits/1a4914c840bfc5fd42ad873a2cd0c66bbf570c7a

File: docs/content/about/projects/_index.md (hunk 4 above)
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md]: Resolved in https://github.com/usnistgov/OSCAL/pull/1702/commits/1a4914c840bfc5fd42ad873a2cd0c66bbf570c7a

File: docs/content/about/projects/_index.md [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+
+Repository: https://github.com/usnistgov/oscal-cli
+
+### OSCAL-content
+
+The OSCAL content repository provides a library of official NIST-maintained OSCAL examples in XML, JSON, and YAML.
+
+Repository: https://github.com/usnistgov/oscal-content
+
+## NIST Dependencies
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md]: Resolved in https://github.com/usnistgov/OSCAL/pull/1702/commits/1a4914c840bfc5fd42ad873a2cd0c66bbf570c7a

File: docs/content/about/projects/_index.md (hunk 2 above)
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:91]: Resolved in https://github.com/usnistgov/OSCAL/pull/1702/commits/1a4914c840bfc5fd42ad873a2cd0c66bbf570c7a

File: docs/content/about/projects/_index.md [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+
+These NIST projects underpin the OSCAL project, but are governed separately from the OSCAL umbrella.
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md]: Resolved in https://github.com/usnistgov/OSCAL/pull/1702/commits/1a4914c840bfc5fd42ad873a2cd0c66bbf570c7a

File: docs/content/about/projects/_index.md [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+    end
+    
+    oscal-->|submodules|metaschema
+    oscal-->|submodules|hugo-uswds
+    liboscal-java-.->|depends on|metaschema-java
+{{</mermaid>}}
+
+## Core Projects
+
+The core projects provide the central functionality or documentation of the OSCAL project and are the primary focus of the NIST OSCAL team.
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md]: Resolved in https://github.com/usnistgov/OSCAL/pull/1702/commits/1a4914c840bfc5fd42ad873a2cd0c66bbf570c7a

File: docs/content/about/projects/_index.md (hunk 1 above)
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md]: Resolved in https://github.com/usnistgov/OSCAL/pull/1702/commits/1a4914c840bfc5fd42ad873a2cd0c66bbf570c7a

**Comment by nikitawootten-nist** [comment]
After integrating your feedback the website now looks like this:

image

**Comment by wendellpiez** [comment]
I like this. It makes me want more exposition and detail. For example, I would like to see where are XSLT dependencies in those repos that have them (OSCAL; metaschema; arguably oscal-content). This could be done in the diagram with some convention (e.g. box outline) but that becomes clutter quickly. Instead, let's think about the next level in, showing more internals about the projects such as platforms, languages, libraries or components, etc.

**Review by iMichaela — CHANGES_REQUESTED** [review]
Please update the graph with the new names/titles.

File: docs/content/about/projects/_index.md [hunk 18]
```diff
@@ -8,10 +8,10 @@ toc:
 
 The OSCAL project is made up of several components, each of which lives in its own repository on GitHub under the [`usnistgov`](https://github.com/usnistgov) organization.
 
-These repositories form a complex patchwork of projects that depend on one-another:
+Where components use code defined in other components, these repositories form a network of dependencies:
 
 {{<mermaid>}}
-graph LR
+graph TD
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply docs/content/about/projects/_index.md:14]: Nice catch, I quickly fixed this in https://github.com/usnistgov/OSCAL/pull/1702/commits/24a222765fa348bb97d348812ca3a0abc65cb6ae

**Review by iMichaela — APPROVED** [review]

**Comment by gregelin** [comment]
@aj-stein-nist You asked on #1649 if I had any comments: 

- It's very nice to see a list of technical projects in one place. 
- Is there a better term than "projects"? These core projects are part of...the OSCAL project" feels awkward. Are these technical projects, tools, or maybe "repos"

I see the OSCAL website is part of the OSCAL project/repo. Could "OSCAL" visually contain "website" and website visually contain "model", "documentation", and "tutorials"? The diagrams for OSCAL models do a nice job of representing hierarchy and maybe a similar pattern would be similar but maybe using circles instead of rectangles to distinguish between models and repositories.

Metaschema and its use deserves more explanation in the above overview, on the documentation page, and in the repo.

**Comment by nikitawootten-nist** [comment]
> I see the OSCAL website is part of the OSCAL project/repo. Could "OSCAL" visually contain "website" and website visually contain "model", "documentation", and "tutorials"? The diagrams for OSCAL models do a nice job of representing hierarchy and maybe a similar pattern would be similar but maybe using circles instead of rectangles to distinguish between models and repositories.

@gregelin: @wendellpiez and I have talked about extending these diagrams to decompose the individual pieces of each repository (maybe in the future we can extend this section with sub-pages for the OSCAL main repository as an example). For now we wanted to keep things relatively high-level, focusing on the relationship between projects/repositories.

**Pull request merged**